code/modules/shuttle/white_ship.dm

```diff
@@ -3,4 +3,429 @@
+	var/area/picked_area = get_area(target_area)
+	if(!src || QDELETED(src))
+		return
+	var/turf/T = safepick(get_area_turfs(picked_area))
+	if(!T)
+		return
+	spawn_atom_to_turf(/obj/effect/overlay/temp/emp/pulse, target_area, 1)
+	landing_zone = new(target_area)
+	landing_zone.id = "engi_ship(\ref[src])"
+	landing_zone.name = "Navigator's Beacon"
+	landing_zone.dwidth = dwidth
+	landing_zone.dheight = dheight
+	landing_zone.width = width
+	landing_zone.height = height
+	landing_zone.setDir(lz_dir)
+	var/list/dock_zone = landing_zone.return_ordered_turfs(landing_zone.x, landing_zone.y, landing_zone.z, landing_zone.dir)
+	for(var/turf/checked in dock_zone)
+		spawn_atom_to_turf(/obj/effect/overlay/temp/emp/pulse, checked, 1)
+
+	for(var/obj/machinery/computer/shuttle/S in machines)
+		if(S.shuttleId == shuttle_id)
+			S.possible_destinations = "[landing_zone.id]"
+	say("Landing Zone Set")
+
+/obj/machinery/computer/engi_nav/proc/GrantActions(mob/living/user)
+	off_action.target = user
+	off_action.Grant(user)
+	beacon.target = user
+	beacon.Grant(user)
+	warp.target = user
+	warp.Grant(user)
+
+
+/mob/camera/aiEye/nav
+	name = "Inactive Navigation Probe"
+	var/sprint = 10
+	var/cooldown = 0
+	var/acceleration = 1
+	var/mob/living/eye_user = null
+	var/obj/machinery/computer/engi_nav/origin
+
+/mob/camera/aiEye/nav/Destroy()
+	eye_user = null
+	origin = null
+	return ..()
+
+/mob/camera/aiEye/nav/GetViewerClient()
+	if(eye_user)
+		return eye_user.client
+	return null
+
+/mob/camera/aiEye/nav/setLoc(T)
+	if(eye_user)
+		if(!isturf(eye_user.loc))
+			return
+		T = get_turf(T)
+		loc = T
+
+/mob/camera/aiEye/nav/relaymove(mob/user,direct)
+	var/initial = initial(sprint)
+	var/max_sprint = 50
+
+	if(cooldown && cooldown < world.timeofday) // 3 seconds
+		sprint = initial
+
+	for(var/i = 0; i < max(sprint, initial); i += 20)
+		var/turf/step = get_turf(get_step(src, direct))
+		if(step)
+			src.setLoc(step)
+
+	cooldown = world.timeofday + 5
+	if(acceleration)
+		sprint = min(sprint + 0.5, max_sprint)
+	else
+		sprint = initial
+
+/datum/action/innate/nav_off
+	name = "End Navigation View"
+	button_icon_state = "camera_off"
+
+/datum/action/innate/nav_off/Activate()
+	if(!target || !isliving(target))
+		return
+	var/mob/living/C = target
+	var/mob/camera/aiEye/nav/N = C.remote_control
+	N.origin.beacon.Remove(C)
+	N.origin.warp.Remove(C)
+	N.eye_user = null
+	if(C.client)
+		C.reset_perspective(null)
+		C.client.change_view(7)
+	C.remote_control = null
+	C.unset_machine()
+	src.Remove(C)
+	playsound(N.origin, 'sound/machines/terminal_off.ogg', 25, 0)
+
+/datum/action/innate/nav_dock
+	name = "Select Docking Location"
+	button_icon_state = "Judicial Marker"
+
+/datum/action/innate/nav_dock/Activate()
+	if(!target || !isliving(target))
+		return
+	var/mob/living/C = target
+	var/mob/camera/aiEye/nav/N = C.remote_control
+	var/obj/machinery/computer/engi_nav/O = N.origin
+	O.SetLanding(N)
+
+/datum/action/innate/z_switch
+	name = "Switch Z Level"

```

Comment:
I hate it on gpses too.
Also ICly it isnt split into 255x255 places, except again, because of GPSes, and we can just as easily call them "Sectors" and avoid the ooc term of Z level.